code/game/objects/items/weapons/fishing_rod.dm

```diff
@@ -0,0 +1,130 @@
+#define TIME_TO_CATCH = 80
+
+// -----------------------------
+//         Bait Datums
+// -----------------------------
+
+// Base datum for bait_type
+// bait: The object to be used as bait
+// fish: list of fish you can catch and their probability
+//       uses the format list(list(<fish>, <absolute probability>), ...)
+//       NOTE: The fishing rod checks the fish in order.  Don't put a probability 100 fish before the last entry.
+/datum/bait_type
+	var/bait = null
+	var/list/fish = list()
+
+/datum/bait_type/standard_bait
+	bait = /obj/item/weapon/reagent_containers/food/snacks/bait
+	fish = list(
+		list(/obj/item/weapon/reagent_containers/food/snacks/fish, 100)
+		)
+
+/datum/bait_type/clown
+	bait = /obj/item/clothing/mask/gas/clown_hat
+	fish = list(
+		list(/obj/item/weapon/reagent_containers/food/snacks/fish/clown, 100)
+		)
+
+// -----------------------------
+//         Fishing Rod
+// -----------------------------
+
+/obj/item/weapon/fishingrod
+	name = "fishing rod"
+	desc = "Go catch the big one!"
+	icon = 'icons/obj/items.dmi'
+	icon_state = "crowbar" //placeholder
+	item_state = "crowbar" //placeholder
+	hitsound = "sound/weapons/toolhit.ogg" //placeholder
+	flags = FPRINT
+	siemens_coefficient = 1
+	slot_flags = SLOT_BELT
+	force = 3.0
+	throwforce = 3.0
+	throw_speed = 1
+	throw_range = 5
+	w_class = W_CLASS_MEDIUM
+	starting_materials = list(MAT_GLASS = 100)
+	w_type = RECYK_GLASS
+	melt_temperature = MELTPOINT_GLASS
+	origin_tech = Tc_ENGINEERING + "=1"
+	attack_verb = list("smacks", "whacks", "whips", "belts", "lashes")
+	var/busy = 0 //check if in use to stop bait scumming
+	var/obj/item/weapon/hookeditem
+	var/list/fishables = list( //list of atoms that can be fished
+		/obj/machinery/bluespace_pond,
+		/turf/unsimulated/beach/water/deep,
+		)
+	var/list/bait_types = list( // The types of bait this rod is able to use
+		/datum/bait_type/standard_bait,
+		/datum/bait_type/clown,
+		)
+
+/obj/item/weapon/fishingrod/suicide_act(mob/user)
+	to_chat(viewers(user), "<span class='danger'>[user] is whipping \his head open with the [src.name]! It looks like \he took the bait!</span>")
+	return(BRUTELOSS)
+
+/obj/item/weapon/fishingrod/afterattack(var/atom/A, var/mob/user)
```

Comment:
You need to use the third argument, `proximity_flag`, here and then check it to prevent people fishing anything in their view.